nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/NuxeoCorsCsrfFilter.java

```diff
@@ -172,6 +230,99 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         response.sendError(HttpServletResponse.SC_FORBIDDEN, message);
     }
 
+    /**
+     * Check safe method according to RFC 7231 4.2.1.
+     */
+    protected boolean isSafeMethod(String method) {
+        return GET.equals(method) || HEAD.equals(method) || OPTIONS.equals(method) || TRACE.equals(method);
+    }
+
+    /**
+     * Manages the CSRF token.
+     * <p>
+     * This method may return a response with token fetch information or with an error if needed, in which case it will
+     * return {@code true}.
+     *
+     * @return {@code true} if the caller doesn't need to do more work (a response has been sent)
+     * @since 10.3
+     */
+    protected boolean manageCSRFToken(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        if (!csrfTokenEnabled) {
+            log.debug("No CSRF token check configured");
+            return false; // no check to do
+        }
+
+        String method = request.getMethod();
+        String path = request.getServletPath();
+        if (path == null) {
```

Comment:
If only Java had the [Elvis operator](https://en.wikipedia.org/wiki/Elvis_operator).